### ISSUE TITLE
Up-to-date Cl packages that DO NOT WORK

### ISSUE THREAD

**Issue by Aqua1ung** [clearlinux/distribution#2868]
There are quite a few CL bundles and packages that do not seem to work. What is remarkable about them is that they are very much _up-to-date_, hence their being broken appears to be due to their _packaging_ rather than anything else. I am thus attempting to list the ones that I have stumbled upon myself, and I am asking if anyone managed to make them work with a reasonably recent CL version (say post-37000). I am also hoping that whoever packaged and/or ported them to CL could chime back with advice as to what I may be doing wrong.

So, the question goes, did anyone manage to make the following work?:

1. [gnome-remote-desktop](https://github.com/clearlinux/distribution/issues/2370);
2. [xpra](https://clearlinux.org/software/bundle/xpra);
3. [xrdp](https://clearlinux.org/software/bundle/x11-tools);
4. [avahi](https://clearlinux.org/software/bundle/avahi-daemon);
5. [gnome-boxes](https://clearlinux.org/software/bundle/gnome-boxes).

**Comment by fenrus75** [comment]
gnome-remote-desktop is a work in progress (not simple since we cannot use
these inside Intel due to some arcane IT rule)
xrdp -- not heard of that before but could be the same issue; they share a
core library

these two the only thing we do to these is remove RC4 encryption support
since our openssl build does not have RC4.

I've not seen any reports on xpra that I remember, but we don't do anything
to it, just bog standard build process

avahi is a bit complicated; unfortunately some gnome desktop stuff pulls in
the library but by default we do NOT want it listening on the network
(basic security policy)...
and we also do not start the network facing stuff by default, and if I
remember we don't install the configuration file by default either.


gnome-boxes used to, and is supposed to work so if it doesn't, that's a bug
that is relatively new, would need more information

**Comment by fenrus75** [comment]
xrdp not supporting wayland is obviously an issue with the X side going
away.

**Comment by Aqua1ung** [comment]
Thank you for the detailed reply!

1. Yes, please try to make gnome-remote-desktop work, I very much appreciate that;
2. As for gnome-boxes, if you attempt to do an "Install from File," the ["Create" button is greyed-out](https://i.imgur.com/4rZuNnu.png), hence cannot be clicked. Interestingly, this bug manifests on the second and all subsequent attempts to "Install from File," so it may take you more than one try to get this to happen. If you could look into this I'd greatly appreciate it!

**Comment by Aqua1ung** [comment]
> xrdp not supporting wayland is obviously an issue with the X side going away.

As far as xrdp not working, [here's what, among other issues, the xrdp head honcho says](https://github.com/neutrinolabs/xrdp/issues/2599#issuecomment-1500966918), namely you guys "have included `startwm.sh` as-is, without any effort to adapt it to [Clear Linux]."

**Comment by matt335672** [comment]
To clarify, I'm not the xrdp head honcho (but thanks for the complement!). I'm just one of the developers.

XRDP currently uses either a modified Xorg install, or Xvnc to provide its services. If you're planning on removing both of these from Clear you won't be able to run XRDP so you can remove it from the Clear repositories, and just point your users in the direction of GNOME remote desktop, which supports some remote desktop use-cases.

On the other hand, if you're continuing to support Xvnc, between us we might be able to work out a way to get xrdp working with that. It really depends on the use-cases you're planning on supporting with Clear.

Remote/virtual desktops aren't really something that Wayland has considered. Until there's more of a consensus behind how this area will be addressed by Wayland I can't see XRDP moving in this direction, but as I said, I'm not the head honcho.

Hope that's useful. I'm going to close the existing issue on our repo for now, but if you want to look at getting Clear and XRDP working together, please raise a new one.

**Comment by fenrus75** [comment]
I turned xrdp into its own bundle and made it start xfce4 as desktop environment, that's better than "xterm"..

I don't have confidence in a "shell script to start a gnome X session"; starting gnome tends to involve a proper systemd session not a shell script, and I also have no confidence that gnome will keep working in X for much longer to begin with.. xfce4 otoh will keep working

**Comment by matt335672** [comment]
Agreed about xfce4 - that seems like a good solution. Also, GNOME has GNOME remote desktop which meets a subset of the xrdp use-cases for users determined to use GNOME.

**Comment by Aqua1ung** [comment]
@fenrus75: going back to gnome-remote-desktop (g-r-d) server, latest releases do [show some promise](https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/issues/149#note_1733782), such as prompting for credentials, though unfortunately credentials are not accepted even though they are correct. Are you aware of this? Any suggestions?

**Comment by coolydudey60** [comment]
I've tried using xrdp but by default cannot connect; initially it wouldn't get anywhere, but after following these steps I was able to reach the same stage of "login failed".

https://github.com/clearlinux/distribution/issues/2760

**Comment by QushyQushy** [comment]
Is this ever gonna work?

**Comment by fenrus75** [comment]
We would recommend gnome-remote-desktop, not xrdp. xrdp is tied to X11 not
Wayland, and since Wayland is our default, with X11 going away over time..
that's where we'd spend our efforts
